base/sort.jl

```diff
@@ -1298,6 +1298,53 @@ function sort!(A::AbstractArray;
     end
     A
 end
+"""
+    sortperm(A; dims::Integer, alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
+
+    Return a permutation vector `I` that puts `v[I]` in sorted order along the given dimension. The order is specified
+    using the same keywords as [`sort!`](@ref). The permutation is guaranteed to be stable even
+    if the sorting algorithm is unstable, meaning that indices of equal elements appear in
+    ascending order.
+
+    To sort slices of an array, refer to [`sortslices`](@ref).
+
+    # Examples
+    ```jldoctest
+    julia> A = [4 3; 1 2]
+    2×2 Matrix{Int64}:
+     4  3
+     1  2
+    
+    julia> sortperm(A, dims = 1)
+    2×2 Matrix{Int64}:
+     2  4
+     1  3
+    
+    julia> sortperm(A, dims = 2)
+    2×2 Matrix{Int64}:
+     3  1
+     2  4
+```
+"""
+function sortperm(A::AbstractArray; dims::Integer,
+    alg::Algorithm=DEFAULT_UNSTABLE,
+    lt=isless,
+    by=identity,
+    rev::Union{Bool,Nothing}=nothing,
+    order::Ordering=Forward)
```

Comment:
I think the indentation in `sort(A::AbstractVctor; dims:Integer)` is more readable than what we have here. I would prefer these lines to be further indented to distinguish the function signature from the body.